pkg/controllermanager/controller/seed/seed_lifecycle_reconcile.go

```diff
@@ -127,6 +129,25 @@ func (c *livecycleReconciler) Reconcile(ctx context.Context, req reconcile.Reque
 		}
 	}
 
+	// If the gardenlet's client certificate is expired and the seed belongs to a `ManagedSeed` then we reconcile it in
+	// order to re-bootstrap the gardenlet.
+	if seed.Status.ClientCertificateExpirationTimestamp != nil && seed.Status.ClientCertificateExpirationTimestamp.UTC().Before(time.Now().UTC()) {
+		managedSeed := &seedmanagementv1alpha1.ManagedSeed{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      seed.Name,
+				Namespace: v1beta1constants.GardenNamespace,
+			},
+		}
+		if err := c.gardenClient.Client().Get(ctx, client.ObjectKeyFromObject(managedSeed), managedSeed); err != nil && !apierrors.IsNotFound(err) {
+			return reconcileResult(err)
+		} else if err == nil {
+			metav1.SetMetaDataAnnotation(&managedSeed.ObjectMeta, v1beta1constants.GardenerOperation, v1beta1constants.GardenerOperationReconcile)
+			if err2 := c.gardenClient.Client().Status().Update(ctx, managedSeed); err2 != nil {
```

Comment:
It's not the `Status` that we want to update here (and this breaks the entire mechanism because the managed seed is not updated). BTW, perhaps you could consider doing this update via a patch?
```suggestion
			if err2 := c.gardenClient.Client().Update(ctx, managedSeed); err2 != nil {
```

Reply:
Nice catch, thank you!